### ISSUE TITLE
A parameter checking issue in the BatchNorm operator

### ISSUE THREAD

**Issue by Embed-Debuger** [tensorflow/tensorflow#59309]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

Yes

### Source

source

### Tensorflow Version

TF 2.3

### Custom Code

Yes

### OS Platform and Distribution

Ubuntu 18.04

### Python version

3.7

### CUDA/cuDNN version

10.1/7.6.1

### Current Behaviour?

```shell
We found that the implementation of tf.keras.layers.BatchNormalization lacks parameter checks. The specific problem comes from its variance parameter. The reproduction of the error is specifically divided into four steps: 
1. Initialize a BN operator (i.e. source_model), randomly input an input (i.e. data), and get an output (i.e. source_result);
2. Set a constant delta, set it to a negative value, such as -1. Subtract delta from the variance of source_model, and add delta to epsilon to get a new BN operator (i.e. follow_model);
3. Input data to follow_model and get follow_result;
4. Calculate the distance between source_result and follow_result. Theoretically, it should be small or even 0, in practice a very large result can be obtained.
We use the same method to implement in pytorch and mindspore. There is no problem with the output of pytorch, and mindspore prompts that the variance of the BN operator should be in the (0, 1] interval.
```

### Standalone code to reproduce the issue

```shell
from tensorflow.keras.layers import BatchNormalization, Input
from tensorflow.keras.models import Model, clone_model

import os
import re
import numpy as np

def SourceModel(shape):
    x = Input(shape=shape[1:])
    y = BatchNormalization(axis=-1)(x)
    return Model(x, y)

def FollowModel_1(source_model):
    follow_model = clone_model(source_model)
    # 读取参数
    weights = source_model.get_weights()
    weights_names = [weight.name for layer in source_model.layers for weight in layer.weights]
    variance_idx = FindWeightsIdx("variance", weights_names)

    # 变异模型
    # delta = np.random.uniform(-1e-3, 1e-3, 1)[0]
    follow_model.layers[1].epsilon += delta     # 写BN层的epsilon
    weights[variance_idx] -= delta
    follow_model.set_weights(weights)

    return follow_model

def FindWeightsIdx(name, weights_names):
    # find layer index by name
    for idx, names in enumerate(weights_names):
        if re.search(name, names):
            return idx
    return -1


os.environ['TF_FORCE_GPU_ALLOW_GROWTH'] = 'true'
os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"
os.environ["CUDA_VISIBLE_DEVICES"] = "0"

shape = (10, 32, 32, 3)
data = np.random.uniform(-1, 1, shape)
delta = -1

source_model = SourceModel(shape)
follow_model = FollowModel_1(source_model)

source_result = source_model(data)
follow_result = follow_model(data)
dis = np.sum(abs(source_result-follow_result))

print("delta:", delta, "; dis:", dis)
```

### Relevant log output

```shell
delta: -1 ; dis: 4510.4536
```
</details>

**Comment by SuryanarayanaY** [comment]
Hi @Embed-Debuger ,

Thanks for your observation. The code tested with TF 2.11v and the behaviour is same as mentioned by you.But with tf-nightly(2.12.0-dev20230123) version even with multiple runs of model and the value of `dis` found as `0.0` everytime. Please refer to attached gist [here](https://colab.research.google.com/gist/SuryanarayanaY/f6f1fd8aee5836ac0ad023f3e7ab31fb/59309-nightly-2-11.ipynb#scrollTo=j6SwlZn4yxfh). As it is resolved with tf-nightly we normally considered it as resolved. Please check and confirm.
Thank you!

**Comment by Embed-Debuger** [comment]
> Hi @Embed-Debuger ,
> 
> Thanks for your observation. The code tested with TF 2.11v and the behaviour is same as mentioned by you.But with tf-nightly(2.12.0-dev20230123) version even with multiple runs of model and the value of found as everytime. Please refer to attached gist [here](https://colab.research.google.com/gist/SuryanarayanaY/f6f1fd8aee5836ac0ad023f3e7ab31fb/59309-nightly-2-11.ipynb#scrollTo=j6SwlZn4yxfh). As it is resolved with tf-nightly we normally considered it as resolved. Please check and confirm. Thank you!`dis``0.0`

Thank you for your reply! Glad to hear that this issue has been resolved, I will close this issue. It is worth mentioning that I think this problem may be caused by TF1.x, because I also found this problem on TF1.13. It could be some similar undetected error code.
